Project Lifecycle.md

```diff
@@ -0,0 +1,94 @@
+
+* TC is 5 members or greater.
+* No more than 1/4 of the TC is affiliated with the same employer.
+* Members of the TC live in at least 4 different timezones.
+* The decision making and release process is documented and publicly accessible.
+
+A project may apply to graduate from incubation at any time by calling for a vote in the TSC.
+
+While a project is incubating it is assigned at least 3 [mentors](https://github.com/nodejs/TSC/blob/master/README.md#mentors) who are responsible for working with the project to adopt policies and gain the health and contributorship it will need in order to graduate from incubation.
+
+## Lifecycle
+
+The Foundation shall encourage new Projects and innovation in the community. New Projects enter the Node.js Foundation through a [Proposal](#Proposal).
+
+The project should be considered mature and have a history of releases before applying to enter the foundation.
+
+## Top Level Project and Working Group Requirements
+
+All TLPs and WGs are expected to operate in a transparent manor. Decisions must be made publicly through a documented public process managed by each TLP TC or WG.
+
+All TLPs and WGs must use a participatory decision making process. All TLP TCs must ensure they are accurately representing the WGs in their TLP.
+
+### Security
+
+All projects in the foundation share the same base security policy. The foundation's security team triages issues sent to security@nodejs.org. Top Level Projects, whether in the incubator or not, are expected to maintain a private security repository where the security team can bring project specific issues.
+
+## Top Level Projects
+
+All Top Level Project TCs must follow a [Consensus Seeking](https://en.wikipedia.org/wiki/Consensus-seeking_decision-making) process and are responsible for documenting and keeping up to date their current processes and practices.
+
+Each TLP TC must elect a representative to the Node.js Foundation TSC or vote to abstain from representation on the TSC.
+
+## Applying to join
+
+A proposal to join the Node.js Foundation as a Top Level Project or Working Group must include:
+
+* Introduction and project description.
+* Project history.
+* Any available metrics or even estimates about the user base, ecosystem and community.
+* Project scope.
+* Current governance process.
+* Current contribution process.
+* List of current tools in use by the project (forums, issue trackers, GitHub orgs, etc).
+* Existing IP Policy and relevant intellectual property (trademarks, domain names, etc).
+* List of initial TC members.
+* List of initial Working Groups.
+* Prior to being admitted the project:
+ * Must include [DCO](https://github.com/nodejs/io.js/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-10).
+ * Must include approved license (MIT, Apache2, etc).
+ * Must include a [Code of Conduct](https://github.com/nodejs/io.js/blob/master/CONTRIBUTING.md#code-of-conduct).
```

Comment:
I notice that these requirements are for both TLPs and WGs. Would it make sense to create separate requirements for both?

In the case of WGs, they are sometimes (often?) brand new things. What is the recommendation for them w.r.t. historical points (current members, processes, etc)?

Also, some WGs won't be producing code, so would creative commons make sense as a license for when WGs only produce documentation? I think a DCO still makes sense for them though, since documentation can also be stolen.